main.tf

```diff
@@ -12,33 +12,39 @@ resource "aws_cloudfront_origin_access_identity" "default" {
   comment = "${module.distribution_label.id}"
 }
 
-data "aws_iam_policy_document" "origin" {
-  statement {
-    actions   = ["s3:GetObject"]
-    resources = ["arn:aws:s3:::${aws_s3_bucket.origin.bucket}${var.origin_path}*"]
-
-    principals {
-      type        = "AWS"
-      identifiers = ["${aws_cloudfront_origin_access_identity.default.iam_arn}"]
-    }
-  }
-
-  statement {
-    actions   = ["s3:ListBucket"]
-    resources = ["arn:aws:s3:::${aws_s3_bucket.origin.bucket}"]
+data "template_file" "bucket_policy_file" {
```

Comment:
Do not see the value proposition of using template file over a policy document. Template file does not validate the JSON while the the policy document does.